paper.tex

```diff
@@ -1293,13 +1297,13 @@ \subsection{Verify Attack Convergence}
 have been found necessary \citep{qian2018l2}.
 
 There are few reasonable threat models under which an attacker can
-compute 100 iterations of gradient descent, but not 1000.
+compute 100 iterations of gradient descent, but not 1000. As stated in~\cite{athalye2018obfuscated}, merely increasing the time it takes an adversary to find an adversarial example (e.g., from one second to ten seconds) does not constitute an increase in robustness.
```

Comment:
Increasing the time from 1 second to 10 days would constitute an increase in robustness, in the sense that the model is harder to break. The metric for some adversarial robustness competitions is the mean perturbation size required to fool a network, or the metric is to drive up the the distortion budget. Driving up the computational budget is also valuable.

Reply:
This is a very tricky point to make, as it ends up depending on the computational model used. Computational cost of an attack is pretty much never taken into consideration (in any security context) unless either: (1) the cost can be argued to be exponential (e.g., for breaking crypto primitives); or (2) there is some clear economic model that explains when an attack is profitable or not.

Reply:
> Computational cost of an attack is pretty much never taken into consideration (in any security context)
It is nearly always taken into consideration in a machine learning context. I am not aware of a strong security motivation for \ell_p adversarial robustness, but I am aware of a machine learning motivation for adversarial robustness.

Reply:
I don't quite understand your point here. 

The machine learning motivation for adversarial robustness is mainly about understanding why adversarial examples exist and how to remove them. What would be the machine learning motivation for training models that have no increased robustness but where finding adversarial examples is more computationally difficult? 

This only seems interesting from a security perspective (in which case I agree that something beyond \ell_p norms is needed). But in that case, you'd also need some security argument as to why an adversary wouldn't spend 10 days to find an adversarial example.

Reply:
> The machine learning motivation for adversarial robustness is mainly about understanding why adversarial examples exist and how to remove them.

It is interesting because adversarial training can provide robustness of other sorts (such as robustness to stochastic corruptions irrespective of the model, other types of adversarial attacks such as adversarial weather (forthcoming), and naturally occurring classifier blindspots (forthcoming)). Adversarial training can also teach the network shape and not to rely on cheap textural cues. There is also an AI safety motivation based on Goodhart's law, but that's a separate discussion.

> you'd also need some security argument as to why an adversary wouldn't spend 10 days to find an adversarial example.
They of course would not spend 10 years, hence there is some security in jacking up the computational budget. While previous works should have increased the attacker budget by an order of magnitude, this is not a reason to suggest to dispense with computational concerns altogether.

Reply:
> They of course would not spend 10 years, hence there is some security in jacking up the computational budget.

I fully agree with this. But it is really hard to figure out what the "right" cutoff point should be. The only satisfactory way we've found so far (in crypto) is to say that increasing the attack cost by a polynomial factor is not enough.

I'm worried that if the community agrees that increasing the attacker's cost is a valuable contribution, we'll see a whole bunch of defenses make arguments like "we've increased the cost of attack by 100x or 1000x". First, this is a very difficult argument to make correctly (i.e., just because your implementation of the attack is 100x slower on your platform doesn't mean this is inherent), and second, it is hard to say whether this is really meaningful or not without context.

Reply:
> we've increased the cost of attack by 100x or 1000x
Defense papers ought not choose to make the attacker weaker than the baseline attacker.

> you still care about the existence of adversarial conditions rather than the computational cost of finding them

If the computational cost is in any way tied to the frequency of occurrence of such conditions, the computation cost is relevant. (I should note that all my responses so far are colored by a speculated defense in which any attacker with a fixed distortion budget is as successful as an attacker randomly sampling perturbations within the budget.)

Reply:
> If the computational cost is in any way tied to the frequency of occurrence of such conditions (imagine that an adversarial distortion is highly influenced by the initialization point), then computational cost becomes relevant.

I agree with this, and if you can argue that in a specific setting, that might be fine. The big difference here is that you would be tying the computational cost to a specific measurable quantity (e.g., the success-rate of a random attacker as in your speculative example).